### PR TITLE
manage users in snowflake

### PR DESCRIPTION
**Pull request by ryanking**

### CODE REVIEW

File: pkg/resources/user.go [hunk 1]
```diff
@@ -0,0 +1,194 @@
...
+
+	return ReadUser(data, meta)
+}
+
+func ReadUser(data *schema.ResourceData, meta interface{}) error {
+	db := meta.(*sql.DB)
+	id := data.Id()
+
+	row := db.QueryRow(fmt.Sprintf("SHOW USERS LIKE '%s'", id))
+	var name, createdOn, loginName, displayName, firstName, lastName, email, minsToUnlock, daysToExpiry, comment, disabled, mustChangePassword, snowflakeLock, defaultWarehouse, defaultNamespace, defaultRole, extAuthnDuo, extAuthnUid, minsToBypassMfa, owner, lastSuccessLogin, expiresAtTime, lockedUntilTime, hasPassword, hasRsaPublicKey sql.NullString
```
[the remark this answers is not in the document]
Comment (edulop91) [reply pkg/resources/user.go]: at my previous job we built/used https://github.com/blend/go-sdk/tree/master/db to help with this kind of things

[the remark this answers is not in the document]
Comment (edulop91) [reply pkg/resources/user.go]: not sure if it is helpful here

File: pkg/resources/user.go [hunk 2]
```diff
@@ -0,0 +1,194 @@
...
+			}
+		}
+	}
+	err = DBExec(db, sb.String())
+
+	if err != nil {
+		return errors.Wrap(err, "error creating user")
+	}
+
+	data.SetId(name)
```
[the remark this answers is not in the document]
Comment (ryanking) [reply pkg/resources/user.go]: its supposed to be

File: pkg/resources/user.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (ryanking) [reply pkg/resources/user.go]: Yeah I agree. I am thinking that when I get a moment I should see if there are any ORM-y libraries that can marshal into and out of structs.

I looked a bit at some of the more popular, but they didn't seem set up to handle DDL queries, so I backed off fearing that I'd have to write a fair bit of code.

**Pull request merged**